### PR TITLE
chore(): cleanup logs and error messages

### PR DESCRIPTION
**Pull request by ShaMan123**
## Motivation

After another dev wrote about the duplicate canvas init error I uderstood the message was not good enough so I decided to open a patch for that.
Then I decided to have all logs and errors in one place, I remember @asturur said something about that.

## Changes

- exposed a `log` internal util
- exposed Error subclasses
- added lint to force usage

### CODE REVIEW

**Review by ShaMan123 — COMMENTED** [review]
ready

**Review by ShaMan123 — COMMENTED** [review]
ok

File: src/canvas/DOMManagers/StaticCanvasDOMManager.ts [hunk 1]
```diff
@@ -33,11 +34,9 @@ export class StaticCanvasDOMManager {
           (getFabricDocument().getElementById(arg0) as HTMLCanvasElement)) ||
         createCanvasElement();
     if (el.hasAttribute('data-fabric')) {
-      /* _DEV_MODE_START_ */
-      throw new Error(
-        'fabric.js: trying to initialize a canvas that has already been initialized'
+      throw new FabricError(
+        'Trying to initialize a canvas that has already been initialized. Did you forget to dispose the canvas?'
       );
-      /* _DEV_MODE_END_ */
```
Comment (asturur) [inline src/canvas/DOMManagers/StaticCanvasDOMManager.ts:40]: i will still do this one day. I can re-add them later

**Review by asturur — DISMISSED** [review]

**Comment by asturur** [comment]
Found 68 of the extra 192 bytes, i m won't look any longer but from the diff i was expecting from the diffs this could easily be a pr that was removing some payload.

**Review by asturur — APPROVED** [review]

**Pull request merged**